### DISCUSSION TITLE
Problem while installing baremetal agw on server pc

### DISCUSSION THREAD

**Discussion by thisisbaskara** [magma/magma#15414] [General]
Dear all,
I kindly seek your assistance in installing Baremetal Magma AGW from the Magma documentation on my HPE Proliant Microserver running Ubuntu Server 20.04. Despite following all the prerequisites, I have not been successful in the installation process, and every time I run `service magma@* status`, I do not get any response.
Furthermore, I have checked the logs using `journalctl -fu agw_installation` and encountered the following error:
`Apr 18 12:22:53 magma bash [27462]:2024-4-18 12:22:53 (61.7 MB/s) - '/root/agw_post_install_ubuntu.sh' saved [4551/4551]`
`Apr 18 12:22:53 magma bash [825]: + reboot`
`Apr 18 12:22:53 magma systemd[1]: agw_installation.service: Main process exited, code killed, status=15/TERM`
`Apr 18 12:22:53 magma systemd[1]: aqw installation.service: Failed with result 'signal'.`
`Apr 18 12:22:53 magma systemd[1]: Stopped AGW Installation.`
I am using the username `magma` and password `magma`. I would appreciate any guidance or assistance that you may offer in resolving this issue.
Thank you for your time and support.
Best regards.

**Comment by tmdzk** [comment]
I can help with that, Let me know when you have time to debug.
Can you drop me the output of
https://github.com/magma/magma/blob/master/lte/gateway/deploy/agw_pre_check_ubuntu.sh
and
https://github.com/magma/magma/blob/master/lte/gateway/deploy/agw_post_install_ubuntu.sh

**Reply by javieraubert** [reply]
@thisisbaskara did you got errors on the ansible tasks?

**Reply by thisisbaskara** [reply]
Hi @javieraubert!
thank you for your reply 

I dont get any errors from Ansible tasks, but I have been stuck at this error for a few months until now I still can't resolve it, eventhough I follow the guidance on the Magma website and the other website but still facing this problem 
[logs from baremetal agw installation.txt](https://github.com/magma/magma/files/15109798/logs.from.baremetal.agw.installation.txt)
Those are logs from agw baremetal installation I did it on HPE proliant microserver with Ubuntu server 20.04.2 with more than 2 ethernet interfaces same as the requirements on magma website

**Comment by rajyay** [comment]
In my case, I followed the documentation for v 1.8 from the https://magma.github.io/magma/docs/lte/deploy_install. After running, bash agw_install_ubuntu.sh script new user 'magma' is created but unable to login as password is unknown and interfaces are renamed to eth0 and eth1 but server is not able to obtain ip. Below is the screenshot of agw_pre_check_ubuntu.sh
![IMG_0079](https://github.com/magma/magma/assets/2726718/12b4605f-d474-4b7f-b20a-ce9e0a6767b5)
. can you help me please?

**Reply by javieraubert** [reply]
You need to run all the scripts as root. Can you post the complete output for the installation script, did you have any error on the ansible tasks?

**Reply by rajyay** [reply]
Thank you javieraubert for the reply. Please check this output after running the script agw_install_ubuntu.sh
Checking if the script has been executed by root user
Checking if Ubuntu is installed
- Check if Ubuntu is installed
Ubuntu is installed
- Check for magma user
magma user configured
- Check if both interfaces are named eth0 and eth1
Hit:1 http://us.archive.ubuntu.com/ubuntu jammy InRelease
Hit:2 http://us.archive.ubuntu.com/ubuntu jammy-updates InRelease
Hit:3 http://us.archive.ubuntu.com/ubuntu jammy-backports InRelease
Hit:4 https://download.docker.com/linux/ubuntu jammy InRelease
Hit:5 https://download.virtualbox.org/virtualbox/debian jammy InRelease
Hit:6 https://linuxfoundation.jfrog.io/artifactory/magma-packages focal-1.8.0 InRelease
Hit:7 http://security.ubuntu.com/ubuntu jammy-security InRelease
Reading package lists...
Need to check if both interfaces are named eth0 and eth1
Interfaces name are correct, let's check if network and DNS are up
Server:		8.8.8.8
Address:	8.8.8.8#53

Non-authoritative answer:
Name:	google.com
Address: 172.253.122.100
Name:	google.com
Address: 172.253.122.138
Name:	google.com
Address: 172.253.122.101
Name:	google.com
Address: 172.253.122.102
Name:	google.com
Address: 172.253.122.139
Name:	google.com
Address: 172.253.122.113
Name:	google.com
Address: 2607:f8b0:4004:c09::8b
Name:	google.com
Address: 2607:f8b0:4004:c09::65
Name:	google.com
Address: 2607:f8b0:4004:c09::8a
Name:	google.com
Address: 2607:f8b0:4004:c09::64

PING 8.8.8.8 (8.8.8.8) from 10.192.232.165 eth0: 56(84) bytes of data.
64 bytes from 8.8.8.8: icmp_seq=1 ttl=115 time=2.48 ms

--- 8.8.8.8 ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 2.479/2.479/2.479/0.000 ms
Making sure magma user is sudoers
Checking if magma has been installed
Magma not installed, processing installation
Reading package lists...
Building dependency tree...
Reading state information...
Package python-dev is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  python2-dev:i386 python2:i386 python2-dev python2 python-dev-is-python3

Requirement already satisfied: ansible==5.10.0 in /usr/local/lib/python3.10/dist-packages (5.10.0)
Requirement already satisfied: ansible-core~=2.12.7 in /usr/local/lib/python3.10/dist-packages (from ansible==5.10.0) (2.12.10)
Requirement already satisfied: cryptography in /usr/lib/python3/dist-packages (from ansible-core~=2.12.7->ansible==5.10.0) (3.4.8)
Requirement already satisfied: jinja2 in /usr/local/lib/python3.10/dist-packages (from ansible-core~=2.12.7->ansible==5.10.0) (3.1.3)
Requirement already satisfied: resolvelib<0.6.0,>=0.5.3 in /usr/local/lib/python3.10/dist-packages (from ansible-core~=2.12.7->ansible==5.10.0) (0.5.4)
Requirement already satisfied: packaging in /usr/local/lib/python3.10/dist-packages (from ansible-core~=2.12.7->ansible==5.10.0) (24.0)
Requirement already satisfied: PyYAML in /usr/lib/python3/dist-packages (from ansible-core~=2.12.7->ansible==5.10.0) (5.4.1)
Requirement already satisfied: MarkupSafe>=2.0 in /usr/lib/python3/dist-packages (from jinja2->ansible-core~=2.12.7->ansible==5.10.0) (2.0.1)
Branch 'v1.8' set up to track remote branch 'v1.8' from 'origin'.
Generating localhost hostfile for Ansible

PLAY [Deploy Magma] ************************************************************

TASK [Gathering Facts] *********************************************************
ok: [127.0.0.1]

TASK [magma_deploy : Include vars of all.yaml] *********************************
ok: [127.0.0.1]

TASK [magma_deploy : Update ca-certs] ******************************************
ok: [127.0.0.1]

TASK [magma_deploy : Forbid usage of expired Let's Encrypt CA cert] ************
ok: [127.0.0.1]

TASK [magma_deploy : Trigger update of CA certs] *******************************
changed: [127.0.0.1]

TASK [magma_deploy : Configuring private package manager.] *********************
ok: [127.0.0.1]

TASK [magma_deploy : Download key] *********************************************
ok: [127.0.0.1]

TASK [magma_deploy : Ignore server ssl cert] ***********************************
ok: [127.0.0.1]

TASK [magma_deploy : Add validated Apt signing key.] ***************************
skipping: [127.0.0.1]

TASK [magma_deploy : Update and upgrade apt packages] **************************
changed: [127.0.0.1]

TASK [magma_deploy : Install ifupdown dependency.] *****************************
ok: [127.0.0.1]

TASK [magma_deploy : Install runtime dependencies.] ****************************
ok: [127.0.0.1]

TASK [magma_deploy : Preconfigure wireshark (tshark) SUID property] ************
changed: [127.0.0.1]

TASK [magma_deploy : Installing magma.] ****************************************
fatal: [127.0.0.1]: FAILED! => {"cache_update_time": 1714065456, "cache_updated": false, "changed": false, "msg": "'/usr/bin/apt-get -y -o \"Dpkg::Options::=--force-confold\" -o \"Dpkg::Options::=--force-confdef\" -o \"Dpkg::Options::=--force-overwrite\"       install 'magma'' failed: E: Unable to correct problems, you have held broken packages.\n", "rc": 100, "stderr": "E: Unable to correct problems, you have held broken packages.\n", "stderr_lines": ["E: Unable to correct problems, you have held broken packages."], "stdout": "Reading package lists...\nBuilding dependency tree...\nReading state information...\nSome packages could not be installed. This may mean that you have\nrequested an impossible situation or if you are using the unstable\ndistribution that some required packages have not yet been created\nor been moved out of Incoming.\nThe following information may help to resolve the situation:\n\nThe following packages have unmet dependencies:\n magma : Depends: python3-pylint (>= 2.6.0) but it is not going to be installed\n         Depends: python3-pylint (>= 2.6.0) but it is not going to be installed\n         Depends: python-redis but it is not installable\n         Depends: libfolly-dev but it is not going to be installed\n         Depends: libprotobuf17 (>= 3.0.0) but it is not installable\n         Depends: libopenvswitch (>= 2.15.4-9-magma) but it is not going to be installed\n", "stdout_lines": ["Reading package lists...", "Building dependency tree...", "Reading state information...", "Some packages could not be installed. This may mean that you have", "requested an impossible situation or if you are using the unstable", "distribution that some required packages have not yet been created", "or been moved out of Incoming.", "The following information may help to resolve the situation:", "", "The following packages have unmet dependencies:", " magma : Depends: python3-pylint (>= 2.6.0) but it is not going to be installed", "         Depends: python3-pylint (>= 2.6.0) but it is not going to be installed", "         Depends: python-redis but it is not installable", "         Depends: libfolly-dev but it is not going to be installed", "         Depends: libprotobuf17 (>= 3.0.0) but it is not installable", "         Depends: libopenvswitch (>= 2.15.4-9-magma) but it is not going to be installed"]}

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=12   changed=3    unreachable=0    failed=1    skipped=1    rescued=0    ignored=0   

Cleanup temp files
AGW installation is done, Run agw_post_install_ubuntu.sh install script after reboot to finish installation

**Comment by javieraubert** [comment]
Hi, sorry for the late reply. Please, start from a fresh ubuntu installation and before running the script can you install this packages:
dpkg --configure -a
apt-get -y install curl make virtualenv zip rsync git software-properties-common python3-pip python-dev apt-transport-https
pip3 install ansible==5.10.0

Because some packages missing your installation is failing. You can see one task from ansible is failing.
